### PR TITLE
CDT-84 Created External Driver access.

### PR DESCRIPTION
**Pull request by bhavinbr**

### CODE REVIEW

File: src/main/java/com/lithium/mineraloil/selenium/elements/DriverManager.java [hunk 1]
```diff
@@ -59,6 +59,10 @@ public void startDriver() {
         log.info("User Agent: " + getUserAgent());
     }
 
+    public void useDriver(WebDriver driver) {
+       DriverInstance driverInstance = new DriverInstance(driver);
+    }
```
Comment (hmcgowan) [inline src/main/java/com/lithium/mineraloil/selenium/elements/DriverManager.java]: look at startDriver. This method needs to do the same thing except the constructor

File: src/test/java/com/lithium/mineraloil/selenium/elements/DriverManagerTest.java [hunk 2]
```diff
@@ -77,4 +79,10 @@ public void openMultipleBrowserWindows() {
         Assertions.assertThat(DriverManager.INSTANCE.getNumberOfDrivers()).isEqualTo(1);
         Assertions.assertThat(DriverManager.INSTANCE.getCurrentUrl()).isEqualTo(testUrl);
     }
+
+    @Test
+    public void useWebDriverInstance() {
+        WebDriver driver = new FirefoxDriver();
+        DriverManager.INSTANCE.useDriver(driver);
```
Comment (hmcgowan) [inline src/test/java/com/lithium/mineraloil/selenium/elements/DriverManagerTest.java]: you need an assertion. Without one, there's no test

File: src/test/java/com/lithium/mineraloil/selenium/elements/DriverManagerTest.java [hunk 3]
```diff
@@ -77,4 +79,12 @@ public void openMultipleBrowserWindows() {
         Assertions.assertThat(DriverManager.INSTANCE.getNumberOfDrivers()).isEqualTo(1);
         Assertions.assertThat(DriverManager.INSTANCE.getCurrentUrl()).isEqualTo(testUrl);
     }
+
+    @Test
+    public void useWebDriverInstance() {
+        DriverManager.INSTANCE.useDriver(DriverManager.INSTANCE.getDriver());
```
Comment (hmcgowan) [inline src/test/java/com/lithium/mineraloil/selenium/elements/DriverManagerTest.java]: this is not right. You should create a new driver and pass it in. Like new ChromeDriver()

File: src/test/java/com/lithium/mineraloil/selenium/elements/DriverManagerTest.java [hunk 4]
```diff
@@ -77,4 +77,12 @@ public void openMultipleBrowserWindows() {
...
+        DriverManager.INSTANCE.get(testUrl);
+        Assertions.assertThat(DriverManager.INSTANCE.getTitle()).isEqualTo("Test");
+        Assertions.assertThat(DriverManager.INSTANCE.getCurrentUrl()).isEqualTo(testUrl);
```
Comment (hmcgowan) [inline src/test/java/com/lithium/mineraloil/selenium/elements/DriverManagerTest.java]: this assertion doesn't check that we're using the driver
a better one to do is DriverManager.INSTANCE.getDriver is equal to the driver you passed in

File: src/main/java/com/lithium/mineraloil/selenium/elements/DriverManager.java [hunk 5]
```diff
@@ -59,6 +59,12 @@ public void startDriver() {
         log.info("User Agent: " + getUserAgent());
     }
 
+    public void useDriver(WebDriver driver) {
```
Comment (hmcgowan) [inline src/main/java/com/lithium/mineraloil/selenium/elements/DriverManager.java]: need the preconditions check

Comment (bhavinbr) [inline src/main/java/com/lithium/mineraloil/selenium/elements/DriverManager.java]: Done

**Pull request merged**